packages/frontend/components/scaffold-eth/Contract/ContractUI.tsx

```diff
@@ -0,0 +1,53 @@
+import { Contract } from "ethers";
+import { useContract, useNetwork, useProvider } from "wagmi";
+import {
+  getAllContractFunctions,
+  getContractReadOnlyMethodsWithParams,
+  getContractVariablesAndNoParamsReadMethods,
+  getGeneratedContract,
+} from "./utils";
+
+type ContractProps = {
+  contractName: string;
+};
+
+/**
+ * TODO
+ * handle error
+ * handle types correctly
+ * handle loading state
+ * handle payable, public, functions
+ **/
+const ContractUI = ({ contractName }: ContractProps) => {
+  const { chain } = useNetwork();
+  const provider = useProvider();
+
+  const { address: contractAddress, abi: contractABI } = getGeneratedContract(chain?.id.toString(), contractName);
+
+  const contract: Contract = useContract({
+    addressOrName: contractAddress,
+    contractInterface: contractABI,
+    signerOrProvider: provider,
+  });
+
+  const displayedContractFunctions = getAllContractFunctions(contract);
```

Comment:
Should we remove this? and maybe use it internally in `getContractVariablesAndNoParamsReadMethods ` and `getContractReadOnlyMethodsWithParams ` functions?

Reply:
I guess if we move it, we'd have to call it three times, right? Inside:

1. getContractVariablesAndNoParamsReadMethods
2. getContractReadOnlyMethodsWithParams
3. (to-do) getContractWriteMethods

In any case, shouldn't be too expensive to call that function 3 times.... so up to you if you want to move it for the sake of readability.

Reply:
Ohh yes, you are right will think about this